examples/control_character/run.py

```diff
@@ -0,0 +1,193 @@
+
+        # initialize input component
+        instance.unreal_service.call_function(
+            uobject=bp_third_person_actor1,
+            ufunction=create_input_component_func,
+            args={"InputComponentToCreate": spear.to_ptr(enhanced_input_component_uclass)})
+        input_component1 = instance.unreal_service.get_component_by_class(bp_third_person_actor1, enhanced_input_component_uclass)
+        instance.input_service.setup_player_input_component(actor=bp_third_person_actor1, input_component=input_component1)
+
+        instance.unreal_service.call_function(
+            uobject=bp_third_person_actor2,
+            ufunction=create_input_component_func,
+            args={"InputComponentToCreate": spear.to_ptr(enhanced_input_component_uclass)})
+        input_component2 = instance.unreal_service.get_component_by_class(bp_third_person_actor2, enhanced_input_component_uclass)
+        instance.input_service.setup_player_input_component(actor=bp_third_person_actor2, input_component=input_component2)
+
+        # get all bone names
+        skeletal_mesh_component1 = instance.unreal_service.get_component_by_class(actor=bp_third_person_actor1, uclass=skeletal_mesh_component_uclass)
+        instance.unreal_service.call_function(
+            uobject=skeletal_mesh_component1,
+            ufunction=set_skeletal_mesh_asset_func,
+            args={"NewMesh": spear.func_utils.to_ptr(manny_simple_uobject)})
+        actor1_bone_names = []
+        skeletal_mesh_component_num_bones1 = instance.unreal_service.call_function(
+            uobject=skeletal_mesh_component1,
+            ufunction=get_num_bones_func)['ReturnValue']
+        for i in range(skeletal_mesh_component_num_bones1):
+            actor1_bone_names.append(instance.unreal_service.call_function(
+                uobject=skeletal_mesh_component1,
+                ufunction=get_bone_name_func,
+                args={"BoneIndex": i})['ReturnValue'])
+
+    with instance.end_frame():
+        pass
+
+    df = pd.DataFrame()
+
+    with instance.begin_frame():
+        instance.unreal_service.call_function(uobject=gameplay_statics_default_object, ufunction=set_game_paused_func, args={"bPaused": False})
+
+        instance.enhanced_input_service.inject_input_for_actor(
+            actor=bp_third_person_actor1,
+            input_action_name="IA_Jump",
+            trigger_event="Started",
+            input_action_value={},
+            input_action_instance={})
+        instance.enhanced_input_service.inject_input_for_actor(
+            actor=bp_third_person_actor2,
+            input_action_name="IA_Jump",
+            trigger_event="Started",
+            input_action_value={},
+            input_action_instance={})
+
+    with instance.end_frame():
+        instance.unreal_service.call_function(uobject=gameplay_statics_default_object, ufunction=set_game_paused_func, args={"bPaused": True})
```

Comment:
- Unpause as the first operation in `begin_frame()`, and pause as the last operation in `end_frame()`, in all of these code blocks for consistency and readability.